sdk/src/main/java/dev/cdevents/CDEvents.java

```diff
@@ -116,35 +109,114 @@ public static CDEvent cdEventFromJson(String cdEventJson) {
     }
 
     /**
-     * Validates the cdEventJson against the Schema URL.
-     * @param cdEventJson
-     * @return true, If cdEventJson is valid
+     * Creates a CloudEvent from the custom cdEvent.
+     * @param <T> customCDEvent class
+     * @param customCDEvent custom CDEvent class object of type <T>
+     * @param validateContextSchema true If validation needed against context.schemaUri
+     * @return CloudEvent
      */
-    public static boolean validateCDEventJson(String cdEventJson) {
-        String eventType = getUnVersionedEventTypeFromJson(cdEventJson);
-        CDEventConstants.CDEventTypes cdEventType = getCDEventTypeEnum(eventType);
+    public static <T extends CDEvent> CloudEvent customCDEventAsCloudEvent(T customCDEvent, boolean validateContextSchema) {
+        if (!validateCustomCDEvent(customCDEvent, validateContextSchema)) {
+            throw new CDEventsException("Custom CDEvent validation failed.");
```

Comment:
Do we not say why something has failed in the exception? We should probably create an issue for that if that's the case.

Reply:
the method with in the condition logs the reason for failures before the exception is thrown.